plugins/out_stackdriver/stackdriver.c

```diff
@@ -402,8 +403,10 @@ static int process_local_resource_id(const void *data, size_t bytes,
         local_resource_id = get_str_value_from_msgpack_map(map, LOCAL_RESOURCE_ID_KEY,
                                                            LEN_LOCAL_RESOURCE_ID_KEY);
         if (!local_resource_id) {
-            msgpack_unpacked_destroy(&result);
-            return -1;
+            /* if local_resource_id is not found, use the current tag of the log */
+            flb_plg_debug(ctx->ins, "local_resource_id not found, "
+                                    "tag [%s] is assigned for local_resource_id", tag);
+            local_resource_id = flb_sds_create(tag);
```

Comment:
What if tag is missing?

Reply:
I think each log event must contain a log. As official doc said: The Name is mandatory and it let Fluent Bit know which input plugin should be loaded. The Tag is mandatory for all plugins except for the input forward plugin (as it provides dynamic tags).

Reference: https://docs.fluentbit.io/manual/v/0.12/configuration/file#config_input